### PR TITLE
Extending showcase route

### PR DESCRIPTION
**Pull request by anuveyatsu**
This PR allows devs to have predefined `datapackage` object.

### CODE REVIEW

File: routes/dms.js [hunk 1]
```diff
@@ -161,10 +161,12 @@ module.exports = function () {
   })
 
   router.get('/:owner/:name', async (req, res, next) => {
-    let datapackage = null
+    let datapackage = res.locals.datapackage || null
```
Comment (starsinmypockets) [inline routes/dms.js]: Is it documented how res.locals.datapackage should used? I assume that this allows themes / plugins to provide their own datapackage by interceptingthe route via `app.param`

Comment (anuveyatsu) [reply routes/dms.js]: @starsinmypockets yes, you also can do it via `app.get('/:owner/:name, ...)` in a theme. I can provide an example: https://gitlab.com/datopian/eds/frontend-eds/blob/master/index.js#L49-61

Comment (starsinmypockets) [reply routes/dms.js]: @anuveyatsu This looks fine to merge. Can we document the use of locals here http://tech.datopian.com/frontend

Comment (anuveyatsu) [reply routes/dms.js]: @starsinmypockets just to confirm, I'm working on docs for this

Comment (anuveyatsu) [reply routes/dms.js]: @starsinmypockets docs are here:
http://tech.datopian.com/frontend/theming/#extending-core-controller

**Pull request merged**